### PR TITLE
Wire mixer.runtime to dynamic.Handler

### PR DESCRIPTION
**Pull request by mandarjog**
Enable grpc adapter dispatch thru Mixer.

### CODE REVIEW

File: mixer/pkg/adapter/test/integration.go [hunk 1]
```diff
@@ -167,7 +172,14 @@ func RunTest(
 	if adapterInfo != nil {
 		adapterInfos = append(adapterInfos, adapterInfo)
 	}
-	if args, err = getServerArgs(scenario.Templates, adapterInfos, scenario.Configs); err != nil {
+	cfgs := scenario.Configs
+	if scenario.GetConfig != nil {
```
Comment (guptasu) [inline mixer/pkg/adapter/test/integration.go]: can you please assert that there must be just one scenaros.Configs or scenario.GetConfig

File: mixer/test/spybackend/nosession_integration_test.go [hunk 2]
```diff
@@ -17,21 +17,181 @@ package spybackend
...
+    instances:
+    - i3list
+---
+`
+
+	i4Quota = `
+apiVersion: "config.istio.io/v1alpha2"
+kind: instance
+metadata:
+  name: i4quota
```
Comment (guptasu) [inline mixer/test/spybackend/nosession_integration_test.go]: this should be "requestCount" or we change the "requestCount" below to "i4quota"

File: mixer/test/spybackend/nosession_integration_test.go [hunk 3]
```diff
@@ -54,14 +214,155 @@ func TestNoSessionBackend(t *testing.T) {
...
+					},
+				},
+
+				// one report request with request.id to match r4 rule
+				{
+					CallKind: adapter_integration.REPORT,
+					Attrs:    map[string]interface{}{"request.id": "somereqid"},
+				},
+			},
+			GetConfig: func(ctx interface{}) ([]string, error) {
```
Comment (guptasu) [inline mixer/test/spybackend/nosession_integration_test.go]: the static config was not right ? Is there a reason to make it dynamic via GetConfig ?

Comment (mandarjog) [reply mixer/test/spybackend/nosession_integration_test.go]: Handler config has connection config which is only available after the server starts.

Comment (guptasu) [reply mixer/test/spybackend/nosession_integration_test.go]: I think we should get rid of static `Configs` completely then for integration tests. I will do that. All tests can just pass via your GetConfig call back.

File: mixer/pkg/protobuf/yaml/dynamic/handler.go [hunk 4]
```diff
@@ -135,11 +141,18 @@ func (h *Handler) connect() (err error) {
 	return nil
 }
 
-func (h *Handler) handleRemote(ctx context.Context, instanceName string, qr proto.Marshaler,
-	dedupID string, resultPtr interface{}, encodedInstances ...[]byte) error {
-	svc := h.svcMap[instanceName]
+func (h *Handler) handleRemote(ctx context.Context, qr proto.Marshaler,
+	dedupID string, resultPtr interface{}, encodedInstances ...*adapter.EncodedInstance) error {
+	svc := h.svcMap[encodedInstances[0].Name]
```
Comment (guptasu) [inline mixer/pkg/protobuf/yaml/dynamic/handler.go]: any concerns of IOOB

File: mixer/test/spybackend/nosession_integration_test.go [hunk 5]
```diff
@@ -360,6 +369,12 @@ func TestNoSessionBackend(t *testing.T) {
 	)
 }
 
+func verifyResult(ctx interface{}, result *adapter_integration.Result) error {
+
+	fmt.Sprintf("%v", result)
```
Comment (guptasu) [inline mixer/test/spybackend/nosession_integration_test.go]: Can you paste what this is printing in this PR comment?

File: mixer/pkg/protobuf/yaml/dynamic/handler.go (hunk 4 above)
Comment (mandarjog) [reply mixer/pkg/protobuf/yaml/dynamic/handler.go]: handleRemote is an internal function. It is always called with at least one encodedInstance.

**Pull request merged**